### PR TITLE
Add a new slack slash command `/dispatch-list-signals` and allow creation of Snooze filter w/ MFA

### PR DESCRIPTION
**Pull request by wssheldon**

### CODE REVIEW

File: src/dispatch/plugins/dispatch_slack/case/interactive.py [hunk 1]
```diff
@@ -45,13 +64,450 @@
...
+            )
+        except RuntimeError as e:
+            if "Invalid request parameters (username)" in str(e):
+                response = mfa_plugin.instance.send_push_notification(
+                    username=email, type="Are you creating a signal filter in Dispatch?"
+                )
+            else:
+                raise e from None
+
+        if response.get("result") == "allow":
```
Comment (wssheldon) [inline src/dispatch/plugins/dispatch_slack/case/interactive.py:476]: We haven't abstracted Auth response across all base MFA plugin providers

**Comment by wssheldon** [comment]
Need to update documentation w/ new command

File: src/dispatch/plugins/dispatch_slack/fields.py [hunk 2]
```diff
@@ -81,6 +106,41 @@ class TimezoneOptions(DispatchEnum):
...
+    initial_option: dict = None,
+    label: str = "Hour",
+    time_format: str = "%Y-%m-%d %H:%M:%S",
+    **kwargs,
+):
+    """Builds a relative date picker input."""
+    now = datetime.now()
+
+    relative_dates = [
+        {"text": "1 hour", "value": now + timedelta(hours=1)},
```
Comment (kevgliss) [inline src/dispatch/plugins/dispatch_slack/fields.py:121]: Shouldn't we delay the calculation until creation? I'm thinking of a situation where somebody opens a modal and goes and gets a coffee.

File: src/dispatch/plugins/dispatch_slack/case/interactive.py [hunk 3]
```diff
@@ -45,13 +64,525 @@
...
+        if idx != len(signals[start_index : end_index + 1]):  # noqa
+            blocks.extend([Divider()])
+
+    pagination_blocks = [
+        Actions(
+            block_id="pagination",
+            elements=[
+                Button(
+                    text="Previous",
+                    action_id="previous",
```
Comment (wssheldon) [inline src/dispatch/plugins/dispatch_slack/case/interactive.py:274]:
```suggestion
                    action_id=CasePaginateActions.list_signal_previous,
```

File: src/dispatch/plugins/dispatch_slack/case/interactive.py [hunk 4]
```diff
@@ -45,13 +64,525 @@
...
+                    style="danger" if current_page == 0 else "primary",
+                ),
+                Button(
+                    text="Next",
+                    action_id="next",
```
Comment (wssheldon) [inline src/dispatch/plugins/dispatch_slack/case/interactive.py:279]:
```suggestion
                    action_id=CasePaginateActions.list_signal_next,
```

**Review by kevgliss — APPROVED** [review]
lgtm

File: src/dispatch/plugins/dispatch_slack/case/interactive.py [hunk 5]
```diff
@@ -45,13 +64,525 @@
...
+            [
+                Section(
+                    text=signal.name,
+                    accessory=Button(
+                        text="Snooze",
+                        value=button_metadata,
+                        action_id=CaseNotificationActions.snooze,
+                    ),
+                ),
+                Context(elements=[MarkdownText(text=f"{signal.variant}")]),
```
Comment (wssheldon) [inline src/dispatch/plugins/dispatch_slack/case/interactive.py:261]:
```suggestion
                Context(elements=[MarkdownText(text=f"{signal.variant}" if signal.variant else "N/A")]),
```

**Pull request merged**